File: Data Structures & Algorithms/max-water-container/submission-8.java
class Solution {
    public int maxArea(int[] heights) {
        int res = 0;
        int l = 0, r = heights.length - 1;
        while (l < r) {
            int currArea = Math.min(heights[l], heights[r]) * (r - l);
            res = Math.max(res, currArea);

            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if (not s):
            return 0

        charSet = set()
        charSet.add(s[0])
        res = 1
        
        l, r = 0, 1
        while r < len(s):
            while s[r] in charSet:
                charSet.remove(s[l])
                l += 1

            charSet.add(s[r])
            res = max(res, r - l + 1)
            r += 1
        
        return res
            
        